### PR TITLE
pythonPackages.pandas: 1.1.1 -> 1.1.4 & cleanup

### PR DESCRIPTION
**Pull request by drewrisinger**
###### Motivation for this change
https://github.com/pandas-dev/pandas/releases/tag/v1.1.4

Mostly bugfix & regression fixes.

While upgrading, I saw a few things in the nix file that could use some cleanup:
* Add changelog
* sort dependencies
* Convert tests to more-standard ``pytestCheckHook``

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Didn't do ``nix-review`` because of large rebuild

### CODE REVIEW

**Comment by drewrisinger** [comment]
ping @FRidh & maintainer @knedlsepp 
I know this is outside the standard python package update script, but I thought the cleanup & other updates were worth the PR.

File: pkgs/development/python-modules/pandas/default.nix [hunk 1]
```diff
@@ -123,19 +129,17 @@ in buildPythonPackage rec {
...
-    runHook postCheck
   '';
 
-  meta = {
+  meta = with stdenv.lib; {
     # https://github.com/pandas-dev/pandas/issues/14866
     # pandas devs are no longer testing i686 so safer to assume it's broken
     broken = stdenv.isi686;
     homepage = "https://pandas.pydata.org/";
+    changelog = "https://pandas.pydata.org/docs/whatsnew/index.html";
```
Comment (knedlsepp) [inline pkgs/development/python-modules/pandas/default.nix:139]: Wasn't even aware we track these. Just read up in the manual that it "may use expansion to refer to the correct changelog". So we could use:" https://pandas.pydata.org/docs/whatsnew/v${version}.html" here.

**Review by knedlsepp — APPROVED** [review]
Thanks! Looks good IMHO.
We could pin the changelog to the particular version. Not sure what's generally the best idea.

File: pkgs/development/python-modules/pandas/default.nix (hunk 1 above)
Comment (drewrisinger) [reply pkgs/development/python-modules/pandas/default.nix:139]: In general, my preference order of operations when adding changelogs to existing packages has been:
1. Generic user-facing changelog (less likely to break), e.g. on readthedocs/whichever site
2. Version-specific link, user-facing
3. GitHub release, populated with version number
4. Some other changelog if I can find it.
Hence I chose 1 here.

**Comment by FRidh** [comment]
Thank you. I've pushed the commits to https://github.com/NixOS/nixpkgs/pull/101636.

**Pull request closed without merging**